### PR TITLE
Batch estimator w/ parameters

### PR DESCRIPTION
**Pull request by thowell**
Modifications to Batch estimator to simultaneously optimize system parameters.

### CODE REVIEW

File: mjpc/estimators/batch.cc [hunk 1]
```diff
@@ -106,13 +109,19 @@ void Batch::Initialize(const mjModel* model) {
...
+  int nconfig = nv * max_history_;
   int nsensortotal = nsensordata_ * max_history_;
+  int ntotal = nconfig + nparam_;
 
   // state dimensions
   nstate_ = nq + nv + na;
   ndstate_ = 2 * nv + na;
 
+  // problem dimensions
+  nconfig_ = nv * configuration_length_;
```
Comment (erez-tom) [inline mjpc/estimators/batch.cc:121]: you just defined this variable 10 lines above, and now you're changing it?  This is confusing...

File: mjpc/estimators/batch.cc [hunk 2]
```diff
@@ -106,13 +109,19 @@ void Batch::Initialize(const mjModel* model) {
 
   // dimension
   int nq = model->nq, nv = model->nv, na = model->na;
-  int ntotal = nv * max_history_;
+  int nconfig = nv * max_history_;
```
Comment (erez-tom) [inline mjpc/estimators/batch.cc:112]: I understand that "nconfig" is referring to the word "configuration" as in "the pose of the body", but this is somewhat clashing with the common use of "config" meaning "settings".  It's not a big deal, but I'd also consider "nframes" or "npose".

**Review by erez-tom — COMMENTED** [review]
Just a few fly-by comments, we'll discuss more today.  Great stuff!

File: mjpc/estimators/batch.cc (hunk 1 above)
Comment (thowell) [reply mjpc/estimators/batch.cc:121]: Renamed these to _max. We use these temporary variables (`ntotal_max`, `nconfig_max`) to preallocate memory.

File: mjpc/estimators/batch.cc (hunk 2 above)
Comment (thowell) [reply mjpc/estimators/batch.cc:112]: Renamed `nconfig` -> `nvel`.

**Comment by thowell** [comment]
Now includes a full (untested) implementation.

File: mjpc/estimators/batch.cc [hunk 3]
```diff
@@ -104,15 +129,21 @@ void Batch::Initialize(const mjModel* model) {
     sensor_start_index_ += model->sensor_dim[i];
   }
 
-  // dimension
+  // allocatation dimension
```
Comment (thowell) [inline mjpc/estimators/batch.cc:132]: spelling..

File: CMakeLists.txt [hunk 4]
```diff
@@ -65,7 +65,7 @@ findorfetch(
   GIT_REPO
   https://github.com/deepmind/mujoco.git
   GIT_TAG
-  main
+  ac150f7
```
Comment (thowell) [inline CMakeLists.txt:68]: restore to main

**Review by erez-tom — APPROVED** [review]

**Pull request merged**